wideq/core.py

```diff
@@ -59,6 +62,32 @@ def get_wideq_logger() -> logging.Logger:
 LOGGER = get_wideq_logger()
 
 
+@lru_cache(maxsize=5)
+def get_retry_session():
+    # See https://www.peterbe.com/plog/best-practice-with-retries-with-requests
+    # for the source of this retry mechanism
+    num_retries = 100  # fail *eventually*, but effectively retry in perpetuity
+    backoff_factor = 0.5
+    status_forcelist = (500, 502, 504)
+    session = requests.Session()
+    session.headers = {
+        'x-thinq-application-key': APP_KEY,
+        'x-thinq-security-key': SECURITY_KEY,
+        'Accept': 'application/json',
+    }
```

Comment:
I'd prefer to keep the headers aspect separate here… can we leave the headers as part of the requests, below? That will help separate the concerns between (a) communicating with the API in the first place, and (b) generically adding retries.

Reply:
Done.